__tests__/todo.spec.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { render, fireEvent, cleanup } from '@testing-library/react-native';
+import { build, fake } from '@jackfranklin/test-data-bot';
+import { firebase as mockFirebase } from '../src/firebase/config';
+
+import Todo from '../src/screens/todo';
+
+// mockFirebase.firestore().useEmulator('localhost', 8080);
+
+const userDataBuilder = build('User Data', {
+  fields: {
+    uid: fake((f) => f.internet.password()),
+    fullName: fake((f) => f.name.findName()),
+    email: fake((f) => f.internet.email()),
+  },
+});
+const user = userDataBuilder();
+
+const dataBuilder = build('Data', {
+  fields: {
+    word: fake((f) => f.lorem.word()),
+  },
+});
+const data = dataBuilder();
+
+const addItem = (targetInput, itemText) => {
+  fireEvent.changeText(targetInput, itemText);
+  fireEvent(targetInput, 'submitEditing');
+};
+
+afterEach(() => cleanup());
+
+test('user is able to add an item to the to-do list', async () => {
+  //* render the todo list
+  const { getByText, getByPlaceholderText } = render(<Todo user={user} />);
+
+  //* type in the input field & click submits
+  addItem(getByPlaceholderText('What needs to be done?'), data.word);
```

Comment:
The code is for human to read, always remember that.
When I arrive to this line _"What is `data`?"_ I need to go up and look for data and realise it is a to-do. 
Perhaps, why no `todoItem` or something like that?

Reply:
Change `data` to `todoItem` as suggested, it read more nicely